### PR TITLE
Add first-class compound physicochemical properties and processes

### PR DESCRIPTION
**Pull request by Felixmil**
Closes #115

`create_compound()` could not produce a compound whose physicochemical properties or processes were correctly registered, and a compound loaded from a snapshot had no supported way to set them. Reaching for the `parameters` argument silently misfiled the values under "Additional Parameters" instead of the physicochemical property groups PK-Sim expects, with no warning. This PR makes every settable building-block compound property first class, both on the `create_compound()` factory and as writable `Compound` fields, so a compound can be built or mutated without hand-authoring raw lists. The scope is 100% coverage of the settable compound properties, verified against all 75 `osp_models()` templates and the snapshot schema.

## New create_compound() arguments

`create_compound()` gains arguments for all five single-parameter alternative groups (`lipophilicity`, `fraction_unbound`, `solubility`, `intestinal_permeability`, `permeability`), each with its `_unit` and `_name` companions and sensible defaults. Solubility is covered in full: `reference_pH`, `solubility_gain_per_charge`, and a table-based `solubility_table` (a two-column pH/value data frame producing a `TableFormula`), with the scalar and table forms mutually exclusive. pKa is set through a typed list argument (`pKa = list(list(type = "Base", value = 10.02), ...)`), and processes attach through `processes = list(create_process(...))`. Unit validation uses the dimensions PK-Sim actually resolves (`"Log Units"`, `"Concentration (mass)"`, `"Velocity"` for both permeability groups).

The roxygen now states explicitly that `parameters` is for additional/loose parameters only and cannot set physicochemical properties, and points to the dedicated arguments instead.

## Writable Compound fields

The matching `Compound` fields (`$lipophilicity`, `$fraction_unbound`, `$solubility`, `$intestinal_permeability`, `$permeability`, `$pka_types`, `$processes`) are now writable, so a compound loaded from a snapshot can be mutated in place. A bare numeric scalar builds the default alternative; a raw list is stored verbatim (the escape hatch for multiple/named/species alternatives, value origins, or table solubility); `NULL` clears the property. `$permeability` is a brand-new read/write binding (only `$intestinal_permeability` existed before).

## Print fix

Printing a compound created without `is_small_molecule` aborted with "missing value where TRUE/FALSE needed" because the unset flag reads as `NA`. The type line is now skipped when the flag is unset, so such a compound prints cleanly.

## Round-trip fidelity

A compound built with the new arguments, added with `add_compound()` and exported with `export_snapshot()`, reloads with the same `Lipophilicity`, `FractionUnbound`, `Solubility`, `IntestinalPermeability`, `Permeability`, `PkaTypes`, and `Processes`. Reassigning a subset of fields on a loaded compound leaves the untouched sections, `Parameters`, and `CalculationMethods` intact. Table-based solubility round-trips at the JSON level; PK-Sim's import-side table preparation is not run by this package (documented on the argument).

## Example

```r
library(osp.snapshots)

inulin <- create_compound(
  name = "Inulin",
  is_small_molecule = TRUE,
  molecular_weight = 5500,
  lipophilicity = -10,
  fraction_unbound = 1,
  solubility = 9999,
  reference_pH = 7,
  pKa = list(list(type = "Base", value = 10.02))
)

inulin
#> ── Compound: Inulin ────────────────────────────────────────────────
#>
#> ── Basic Properties ──
#> • Type: Small Molecule
#> • Molecular Weight: 5500 g/mol
#>
#> ── Physicochemical Properties ──
#> • Lipophilicity:
#>   • -10 Log Units [Unknown]
#> • Fraction Unbound:
#>   • 1 [Unknown]
#> • Solubility:
#>   • 9999 mg/l (pH 7) [Unknown]
#> • pKa Types:
#>   • Base: 10.02 [Unknown]
```

## Summary by CodeRabbit

* **New Features**
  * Create compounds with additional physicochemical inputs (lipophilicity, fraction unbound, solubility, intestinal permeability, permeability), plus pKa entries and attached processes.
  * Solubility supports scalar inputs (with optional reference pH and gain-per-charge) or pH-based tables.
* **Bug Fixes**
  * Compound printing no longer errors when the small-molecule flag is left unset/NA.
  * Improved handling and validation of physicochemical property assignments (including clearing with `NULL`).
* **Documentation**
  * Updated class and function reference docs and release notes; added snapshot test coverage for edge cases and validation errors.

### CODE REVIEW

File: R/create_compound.R [hunk 1]
```diff
@@ -67,6 +195,52 @@ create_compound <- function(
...
+      "i" = "Supply {.arg solubility} or {.arg solubility_table}, not both."
+    ))
+  }
+  if (!is.null(solubility_table)) {
+    if (!is.data.frame(solubility_table) || ncol(solubility_table) != 2) {
+      cli::cli_abort(
+        "{.arg solubility_table} must be a data frame with two columns (pH, value)"
+      )
+    }
+  }
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply R/create_compound.R:237]: Fixed in cca7f3a. The `solubility_table` guard now also rejects a zero-row table and non-numeric pH/value columns (`nrow == 0 || !is.numeric(col1) || !is.numeric(col2)`), so a character column or empty table aborts before `build_solubility_table_formula()` indexes the columns, rather than serializing a malformed `TableFormula`. Added `expect_snapshot(error = TRUE)` tests for the empty-table and non-numeric-column cases.

File: R/create_compound.R [hunk 2]
```diff
@@ -104,5 +278,184 @@ create_compound <- function(
...
+    data$Permeability <- build_single_param_alternative(
+      permeability_name,
+      "Permeability",
+      permeability,
+      permeability_unit
+    )
+  }
+  if (!is.null(pKa) && length(pKa) > 0) {
+    data$PkaTypes <- build_pka_types(pKa)
+  }
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply R/create_compound.R:341]: Good catch on the mismatch. Resolved in cca7f3a by correcting the documentation rather than the behavior. At construction there is nothing to clear, and emitting an empty `[]` `PkaTypes` array on a fresh compound would be undesirable for PK-Sim, so `create_compound(pKa = list())` and `pKa = NULL` intentionally both leave no `PkaTypes` key. The `@param pKa` roxygen now states that accurately, and clarifies that the clearing semantics (an empty `list()` removing existing values) live on the writable `Compound$pka_types` field, which already treats `NULL` and `list()` identically as "clear".
